tools/xtask/src/versions.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+use anyhow::Context;
+use anyhow::Result;
+use regex::Regex;
+use serde_json::Map;
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use crate::get_cargo_metadata;
+use crate::get_rust_paths;
+
+pub fn version_command() -> anyhow::Result<()> {
+    let workspace_root = crate::get_workspace_root()?;
+
+    let mut versions = HashMap::new();
+    let mut rust_versions = HashMap::new();
+    let mut rust_deps = HashMap::new();
+
+    // Process meson version
+    if let Err(e) = process_meson_version(workspace_root.as_path(), &mut versions) {
```

Comment:
Should we just ignore meson at this point?

Reply:
Yeah, let's ditch it. I was going to keep this as the v1 format matching `version-tool.py` output, but given we (maybe I?) are the only consumers of this script, let's break the format now. I'll rename the entries to not include the numbers too.